### DISCUSSION TITLE
Mapped type "PartialType"

### DISCUSSION THREAD

**Discussion by ByteSturm** [nartc/mapper#496] [Ideas]
Hey @nartc,
in my NestJS project I use for my update DTOs often the mapped type "PartialType" like in the following example:
`export class UpdateCatDto extends PartialType(CreateCatDto) {}`

When looking at the documentation of automapper, I saw that there do exist [mapped types](https://automapperts.netlify.app/docs/misc/mapped-types) too. I know the NestJS "PartialType" is from the `@nestjs/swagger` package. Would it make sense to add this "PartialType" functionality to automapper? (At best compatible to the `@nestjs/swagger` package 😄 )

The benefit would be not having to define the update DTOs explicitly if they are identical as the create DTOs, just all properties being optional.

**Comment by nartc** [comment]
Hey @ByteSturm, it is actually a blocker on my end as I'm not sure how to compose the mixin (the mapped types) in a type-safe way https://github.com/nartc/mapper/issues/271
